students/jesse_miller/session09/donor_models.bak.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+'''
+Let's try this by following the directions.  So, this is going to be where the
+donor manipulation occurs.  The goal is to compartmentalize everything so that
+functions can be called by other programs if and when needed.
+'''
+
+
+class Donor:
+    '''
+    This is not being used at all.  Every time I've attempted to use these
+    functions something has broken in a unique and fun way.  For example, key
+    errors.
+    '''
+
+    def __init__(self, name):
+        self.name = name
+        self.donations = []
+
+
+    def donation_add(self, new_donation):
+        '''
+        This should append donations to the list
+        '''
+        self.donations.append(new_donation)
+
+    @property
+    def donation_count(self):
+        '''
+        Fairly obvious, counts the number of donations
+        '''
+        return len(self.donations)
+
+
+    @property
+    def donation_total(self):
+        '''
+        Again, obvious, adds the donations up
+        '''
+        return sum(self.donations)
+
+
+    @property
+    def donation_average(self):
+        '''
+        Averages the donations
+        '''
+        return self.donation_total / self.donation_count
+
+
+################################################################################
+'''
+End Donors
+'''
+################################################################################
+
+class DonorCollection:
+    '''
+    This class will handle the donors dictionary, as well as the collections.
+    Some of the things we can look forward to:
+    Create a donor
+    Find a donor
+    List donors
+    Donor reports
+    Data saving and reloading.
+    '''
+    def __init__(self):
+        '''
+        Initializing and building the default dict
+        '''
+        self.donors_dict = {'Robert Smith': [435.56, 125.23, 357.10],
```

Comment:
Remove loading sample data for now and try adding new donors manually. The problem with this initialization is that it will cause your _donors_dict_ to have two types of data:

- List of donations (for the sample donors added during initialization).
- Donor objects (added when creating new donors).